services/QuillLMS/db/migrate/20210824114552_create_provider_classroom_user.rb

```diff
@@ -0,0 +1,17 @@
+class CreateProviderClassroomUser < ActiveRecord::Migration[5.1]
+  def change
+    create_table :provider_classroom_users do |t|
+      t.string :type, null: false
+      t.string :provider_classroom_id, index: true, null: false
+      t.string :provider_user_id, index: true, null: false
+      t.timestamp :deleted_at
+
+      t.timestamps
+    end
+
+    add_index :provider_classroom_users,
+      [:provider_user_id, :provider_classroom_id, :type],
```

Comment:
Flagging this to double check that this index order is optimal for common queries.

(As you likely know, order matters in compound indexes i.e. the index can only be used by a query if the left-most index column is in the query (`provider_user_id` here). So, for example, would it be common to query by just `type` and `provider_classroom_id`? If so, we might consider a different order (or different index setup).)

Reply:
This is a good point.  Spoke with Dan about this offline and we agreed on just starting with a `[:type, :provider_classroom_id, :provider_user_id]` compound since `type` is almost always be used, and most queries currently use :provider_classroom_id`.